feat(xo-lite/Console): ability to scale VM console

MathieuRA opened the pull request:
### Check list

> Check if done, if not relevant leave unchecked.

- [ ] PR reference the relevant issue (e.g. `Fixes #007` or `See xoa-support#42`)
- [ ] if UI changes, a screenshot has been added to the PR
- [ ] documentation updated
- `CHANGELOG.unreleased.md`:
  - [ ] enhancement/bug fix entry added
  - [ ] list of packages to release updated (`${name} v${new version}`)
- **I have tested added/updated features** (and impacted code)
  - [ ] unit tests (e.g. [`cron/parse.spec.js`](https://github.com/vatesfr/xen-orchestra/blob/b24400b21de1ebafa1099c56bac1de5c988d9202/%40xen-orchestra/cron/src/parse.spec.js))
  - [ ] if `xo-server` API changes, the corresponding test has been added to/updated on [`xo-server-test`](https://github.com/vatesfr/xen-orchestra/tree/master/packages/xo-server-test)
  - [x] at least manual testing

### Process

1. create a PR as soon as possible
1. mark it as `WiP:` (Work in Progress) if not ready to be merged
1. when you want a review, add a reviewer (and only one)
1. if necessary, update your PR, and re- add a reviewer

From [_the Four Agreements_](https://en.wikipedia.org/wiki/Don_Miguel_Ruiz#The_Four_Agreements):

1. Be impeccable with your word.
1. Don't take anything personally.
1. Don't make assumptions.
1. Always do your best.

pdonias committed:
feat(xo-lite): initial commit

pdonias committed:
feat(xo-lite): initial commit

pdonias committed:
fix(xo-lite/console): get console URL from XAPI

pdonias committed:
fix(xo-lite/console): get console URL from XAPI

pdonias committed:
feat(xo-lite): Reaclette types

pdonias committed:
feat(xo-lite): Reaclette types

julien-f committed:
feat(xapi/VM_create): generateMacSeed option

MathieuRA committed:
feat(xo-server): VM_destroy instead of deleteVm (#5693)

Continuation of 5f1c127

badrAZ committed:
feat(xo-server-usage-report): add VM IP addresses to the report (#5696)

julien-f committed:
fix(backups/importDeltaVm): new mac_seed when using newMacAddresses

julien-f committed:
fix(xo-web/vm/advanced): RTL819 → RTL8139

Fixes #5698

julien-f committed:
fix(xapi/task_create): remove duplicate [XO] prefix

Found while investigating xoa-support#3553

julien-f committed:
fix(xapi/task_create): dont use super

Introduced by 021810201b3829e23f76ef5f212a7c48b8284667

It cannot be used due to our mixin architecture.

julien-f committed:
chore: named import from @xen-orchestra/log

julien-f committed:
chore(log/transport/console): remove unused code

julien-f committed:
feat(xapi): update to xo-common@0.7.0

rajaa212 committed:
fix(xo-web/vm/advanced): remove noop (#5700)

See https://github.com/vatesfr/xen-orchestra/pull/3774/files#r602860699

julien-f committed:
feat(backups/delta writers): split run method in prepare/transfer/cleanup

Fixes xoa-support#3523

This avoids starting the transfer before the writers are ready, which caused it to failed with `deleteFirst` when deletion was so long that the transfer stalled.

julien-f committed:
feat(backups/Task.wrapFn): compatibility with @decorateWith

pdonias committed:
feat: technical release (#5702)

julien-f committed:
chore(proxy): remove unused Node test

The proxy now requires Node >=12

julien-f committed:
feat(xapi/VDI_destroy): dont fail if VDI not found

Aligned with xo-server/xapi/deleteVdi.

julien-f committed:
feat(xo-server): VDI_destroy instead of deleteVdi

julien-f committed:
feat(xapi/call{,Async}): retry if too many pending tasks

Logic from xo-server/xapi/call

julien-f committed:
feat(import VM backup): newMacAddresses setting

Related to #5697

julien-f committed:
fix(xapi/call{,Async}): fix call to retry

Introduced by 3bb7d2c29

MathieuRA committed:
feat(xo-web/proxies): move proxy actions to dropdown (#5688)

julien-f committed:
fix(proxy/api/ndJsonStream): dont fail if one message cannot be JSONified

Wescoeur committed:
feat(load-balancer): add new anti-affinity mode (#5652)

Fixes #5600

rajaa212 committed:
feat(xo-server-perf-alert): ability to choose all hosts, VMs and SRs (#5692)

Fixes #2987

MathieuRA committed:
feat(xo-web/restore): support new mac addresses (#5697)

julien-f committed:
fix(proxy/backup.importVmBackup): settings param is optional

Introduced by b475b265a

julien-f committed:
fix(backups/importDeltaVm): reverse newMacAddresses condition for VIF creation

Introduced by 4df8c9610

MathieuRA committed:
feat(xo-lite): possibility to scale the VM console

MathieuRA committed:
feat(xo-lite): possibility to scale the VM console

MathieuRA committed:
feat(xo-lite): possibility to scale the VM console

MathieuRA committed:
prettier

MathieuRA committed:
prettier

MathieuRA committed:
prettier

MathieuRA committed:
prettier

MathieuRA committed:
prettier

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,25 @@
+  step: number
+}
+const RangeInput: React.FC<PropsRangeInput> = ({ defaultValue, max, min, onChange, step }) => {
+  const [currentValue, setCurrentValue] = React.useState(defaultValue)
```
Use Reaclette instead.

pdonias commented on @xen-orchestra/lite/src/components/Console.tsx:
```diff
@@ -50,13 +52,41 @@ const Console = withState<State, Props, Effects, Computed, ParentState, ParentEf
+    </>
+    )
+
+  }
```
Don't edit `Console.tsx` directly. This component should remain a simple wrapper that exposes novnc's features. Instead, create a (temporary) new page that embeds the console component. Then, that page will be used inside the home page.

badrAZ committed:
feat(fs): expose highWaterMark stream option (#5676)

rajaa212 committed:
feat(xo-server,xo-web/startVm): avoid booting VM if there's an identical MAC address elsewhere (#5655)

Fixes #5601

pdonias committed:
feat: technical release (#5704)

pdonias commented on @xen-orchestra/lite/src/components/Console.tsx:
```diff
@@ -50,13 +52,41 @@ const Console = withState<State, Props, Effects, Computed, ParentState, ParentEf
+      })
+
+      // To resize the console automatically
+      // rfb.resizeSession not working
```
> `resizeSession` is a `boolean` indicating if a request to resize the remote session should be sent whenever the container changes dimensions. Disabled by default.

We don't want to change the VM's screen resolution, we just want to scale the canvas to the dimensions we want. So `scaleViewport` should indeed be the way to go.

pdonias commented on @xen-orchestra/lite/src/components/Console.tsx:
```diff
@@ -50,13 +52,41 @@ const Console = withState<State, Props, Effects, Computed, ParentState, ParentEf
+    return (
+    <>
+      <RangeInput defaultValue={1} max={3} min={0.1} onChange={_scaleConsole} step={0.1} />
+      <div ref={state.container} style={consoleSize}/>
```
Missing space at the end.

pdonias commented on @xen-orchestra/lite/src/components/Console.tsx:
```diff
@@ -50,13 +52,41 @@ const Console = withState<State, Props, Effects, Computed, ParentState, ParentEf
   },
-  ({ state }) => <div ref={state.container} />
+  ({ state }) => {
+    const [consoleSize, setConsoleSize] = React.useState({
```
Use Reaclette.

MathieuRA committed:
RangeInput using reaclette

MathieuRA committed:
RangeInput using reaclette

julien-f committed:
chore(xo-server/getXapiObject): clarify that second param is an XO type

julien-f committed:
fix(xo-server): Backup/getConnectedRecord receive XAPI not XO type

MathieuRA committed:
sort

MathieuRA committed:
sort

MathieuRA committed:
sort

MathieuRA committed:
create TabConsole componant using reaclette

MathieuRA committed:
create TabConsole componant using reaclette

MathieuRA committed:
create TabConsole componant using reaclette

MathieuRA committed:
remove useless code

MathieuRA committed:
remove useless code

MathieuRA committed:
remove useless code

MathieuRA committed:
remove public instead private scaleConsole

MathieuRA committed:
remove public instead private scaleConsole

pdonias commented on @xen-orchestra/lite/src/components/Console.tsx:
```diff
@@ -50,13 +54,14 @@ const Console = withState<State, Props, Effects, Computed, ParentState, ParentEf
         url.searchParams.set('session_id', xapi.sessionId)
 
         // eslint-disable-next-line no-new
```
Remove this line.

pdonias commented on @xen-orchestra/lite/src/components/TabConsole.tsx:
```diff
@@ -0,0 +1,62 @@
+          width: 1024 * value,
+        }
+        // With "scaleViewport", the canvas occupies all
+        // available space of his container.
```
```suggestion
        // available space of its container.
```

pdonias commented on @xen-orchestra/lite/src/components/TabConsole.tsx:
```diff
@@ -0,0 +1,62 @@
+        // With "scaleViewport", the canvas occupies all
+        // available space of his container.
+        // But when the size of the container is changed,
+        // the canvas size aren't updated
```
```suggestion
        // the canvas size isn't updated
```

pdonias commented on @xen-orchestra/lite/src/components/TabConsole.tsx:
```diff
@@ -0,0 +1,62 @@
+    initialState: () => ({
+      consoleSize: {
+        height: 768,
+        width: 1024,
```
Why these specific values?

MathieuRA replied:
This is the default size of the canvas from `novnc`

pdonias replied:
It doesn't have a default size, it has the same dimensions as the VM screen resolution.

MathieuRA replied:
Ho okay. And the VM screen resolution can be different on each VM ?

pdonias replied:
Yes, it can be anything at all depending on the OS settings.

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,59 @@
+        if (current === null) {
+          return
+        }
+        this.state.currentValue = +current.value
```
Use the argument event instead.

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,59 @@
+const RangeInput = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: props => ({
+      currentValue: props.defaultValue,
```
Make it a controlled component instead of handling a default value.

MathieuRA replied:
I don't understand. He is already a controlled component no ? 
The value of the input is equal to the value state `currentValue`.
The `defaultValue` prop is here just to set a first value.

pdonias replied:
I'm talking about `RangeInput`. Right now, `input` is controlled by `RangeInput` but `RangeInput` isn't controlled by its parent.

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,59 @@
+    },
+  },
+  ({ effects, max, min, state, step }) => {
+    return (
```
Remove brackets and `return`.

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,59 @@
+        type='range'
+        value={state.currentValue}
+      />
+    )
```
I think it would be a good idea to forward any other prop to the `input` element. That way we wouldn't have to change the implementation if we want to use, say, a `for` attribute on the input itself.

pdonias commented on @xen-orchestra/lite/src/components/Console.tsx:
```diff
@@ -50,13 +54,14 @@ const Console = withState<State, Props, Effects, Computed, ParentState, ParentEf
     },
   },
-  ({ state }) => <div ref={state.container} />
+  ({ state }) => <div ref={state.container} style={state.consoleSize} />
```
I think it would be better to pass a `scale` prop instead. Also, state inheritance may be removed from Reaclette in the future.

MathieuRA replied:
`effects` inheritance too ?

pdonias replied:
Yes.

julien-f committed:
fix(backups/delta writers): compute old entries before run

julien-f committed:
feat(backups/CR writer): delete interrupted copies in prepare()

pdonias committed:
feat: technical release (#5705)

julien-f committed:
feat: release 5.57.0

julien-f committed:
fix(CHANGELOG): missing Highlights title

julien-f committed:
fix(CHANGELOG): move Highlights title under badge

Introduced by 1e9e78223

julien-f committed:
feat(@xen-orchestra/proxy): 0.12.0

julien-f committed:
feat(xapi): add xo-xapi CLI

Allows to easily test @xen-orchestra/xapi features.

MathieuRA committed:
use % value instead of px value

MathieuRA committed:
use % value instead of px value

MathieuRA committed:
use % value instead of px value

MathieuRA committed:
forward all possible attribute

MathieuRA committed:
forward all possible attribute

MathieuRA committed:
forward all possible attribute

MathieuRA committed:
controlled default value

MathieuRA committed:
controlled default value

MathieuRA committed:
controlled default value

MathieuRA committed:
do not inline props property

MathieuRA committed:
do not inline props property

MathieuRA committed:
do not inline props property

MathieuRA committed:
remove useless bracket

MathieuRA committed:
remove useless bracket

julien-f committed:
chore: update dependencies

julien-f committed:
chore(xo-server-load-balancer): format with Prettier

Wescoeur committed:
feat(load-balancer): add option to disable migration (#5706)

MathieuRA committed:
fix(xo-web/restore): generateNewMACAddresses disabled by default (#5707)

julien-f committed:
chore(CHANGELOG.unreleased): formatting

julien-f committed:
fix(xapi/VM_snapshot): VM#refresh_snapshots does not exist

Fixes xoa-support#3587

Fohdeesha committed:
fix(xo-web/en): grammar fixes (#5713)

julien-f committed:
chore(xo-server/Xo): pass config as named param

As done in xo-proxy.

julien-f committed:
feat(xo-server/Xo): safeMode param

julien-f committed:
feat(xo-server/XenServers): dont auto connect on safe mode

julien-f committed:
feat(xo-server/Xo): pass appDir, AppName and httpServer

As done in xo-proxy.

This may allow sharing mixins in the future.

julien-f committed:
chore(xo-server/http): use @xen-orchestra/defined

julien-f committed:
chore(xo-server/utils): remove unused wrap

julien-f committed:
chore(xo-server/proxies): use @vates/read-chunk

julien-f committed:
feat(read-chunk): handle 0 size

julien-f committed:
chore(read-chunk): add tests

julien-f committed:
feat(read-chunk/package.json): only publish index.js

julien-f committed:
feat(@vates/read-chunk): 0.1.2

julien-f committed:
chore(vhd-lib): use @vates/read-chunk

julien-f committed:
fix(xo-server/proxies): readChunk import

Introduced by 538253cdc

julien-f committed:
fix(compose/USAGE): typo

julien-f committed:
chore(normalize-packages): remove unused require

julien-f committed:
feat: unified .npmignore for all packages

Ensure sources, tests and USAGE.Md files will not be published.

julien-f committed:
fix(.gitignore): handle @vates/* pkgs

julien-f committed:
fix(xapi): add missing cli.js

Introduced by be9b5332d980886b979178cbeefc49216b6ad87d

julien-f committed:
feat(read-chunk): ensure function is properly named

julien-f committed:
chore: add descriptions to all pkgs (but mixin)

julien-f committed:
feat(normalize-packages): delete empty dependencies, description and keywords fields

julien-f committed:
fix(proxy): upgrade to koa-compress@5

Fix a koa-compress@4 bug which compress the reply when no accept-encoding header has been passed.

Darkbeldin committed:
feat(doc): Host update revision (#5716)

* Host update revision

Co-authored-by: yannick Achy <yannick.achy@vates.fr>

julien-f committed:
chore: use @babel/eslint-parser instead of babel-eslint

babel-eslint is no longer maintained and has issues with some recent syntaxes like private methods.

julien-f committed:
chore: remove unnecessary babel-core

julien-f committed:
chore: remove unnecessary Jest transform setting

julien-f committed:
chore: disable unused Jest coverage

julien-f committed:
fix(backups/DeltaBackupWriter): dont overwrite prepare/cleanup in constructor

Introduced in e69b6c4dc

julien-f committed:
chore(backups): remove unused deps

julien-f committed:
feat(backups/Task): new implementation

- no longer requires logging
- supports cancelation (`Task.cancelToken` and `Task#cancel()`)
- supports running multiple functions in the same task

julien-f committed:
fix(backups/DeltaBackupWriter): merge should be subtask of export

Introduced by f5024f0e7

julien-f committed:
feat(backups/Task.wrapFn): opts can be a function

julien-f committed:
fix(xo-server/api): dont log pool.listMissingPatches & host.stats errors

Introduced by 9226c6cac

julien-f committed:
fix(xo-server/api): log instead of rejecting  non-strict schemas

Fixes https://xcp-ng.org/forum/topic/4439/plugin-transport-email-v0-6-0-broken

julien-f committed:
feat(xo-server/api): explicitely allow $type and enumNames in schemas

julien-f committed:
fix(CHANGELOG.unreleased): add missing entry

Related to 3cc9fd278

julien-f committed:
chore(backups/readDeltaVmBackup): remove unused value

julien-f committed:
fix(backups/ImportVmBackup): use transfered size instead of backup size

Backup size is smaller in case of delta VHDs.

julien-f committed:
fix(backups/ImportVmBackup): use transfered size instead of backup size

Backup size is smaller in case of delta VHDs.

Fohdeesha committed:
fix(xo-web/en): more grammar fixes (#5714)

Fohdeesha committed:
fix(xo-web/en): more grammar fixes (#5714)

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,53 @@
+      },
+    },
+  },
+  ({ effects, inputAttribues, state }) => (
```
You don't need to have an extra prop for this, you can use the rest operator to get all the props:
```js
({ effects, state, ...props }) =>
```

MathieuRA replied:
I created an extra prop, because by using the rest operator to get all the props, am getting some props that i dont want.

Some exemples:
i get the `onChange` prop, and it will overide the `onChange` value of the input.
I get the `resetState` prop, and he will try to inject this function into the `input` tag, and generate an error in the web console.

By creating extra prop, this give me more control of what i want inject into the `input` tag and let me decide if i want to omit some `HTML input function` like what i did for `onChange` [here](https://github.com/vatesfr/xen-orchestra/pull/5703/files#diff-8365d3997b0fda0405f43030763b8247b7d07963364eed4bb039ea4d22fefa9eR21).

pdonias replied:
Ok, but when you're using this component, you shouldn't have to know which props go into the `inputAttributes` and which ones need to be passed directly.

If you want to grab some props for your implementation, you can destructure them before the rest operator and then forward everything else to the input:
```js
({ effects, state, resetState, onChange, ...props }) =>
```

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,53 @@
+const RangeInput = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: props => ({
+      currentValue: props.defaultValue,
```
You shouldn't need to maintain an internal state if the input is fully controlled.

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,53 @@
+
+interface Props {
+  defaultValue: number
+  inputAttribues: Omit<InputAttributes, 'onChange'>
```
Typo.

pdonias commented on @xen-orchestra/lite/src/components/TabConsole.tsx:
```diff
@@ -0,0 +1,69 @@
+import Console from './Console'
+import RangeInput from './RangeInput'
+
+const INPUTATTRIBUTES = {
```
`INPUT_ATTRIBUTES`

pdonias commented on @xen-orchestra/lite/src/components/TabConsole.tsx:
```diff
@@ -0,0 +1,69 @@
+        // Value in percent
+        height: 100,
+        width: 100,
+      },
```
You only need one value for this.

julien-f committed:
fix(xo-server/xapi/startVm): dont destructure options without default value

See xoa-support#3613

julien-f committed:
fix(xo-server/xapi/startVm): dont destructure options without default value

See xoa-support#3613

julien-f committed:
feat(xo-server/xapi/startVm): move hostId into options

julien-f committed:
feat(xo-server/xapi/startVm): move hostId into options

julien-f committed:
fix(CHANGELOG.unreleased): add missing entriy

Introduced by 4bbe8488f

julien-f committed:
fix(CHANGELOG.unreleased): add missing entriy

Introduced by 4bbe8488f

julien-f committed:
fix(xo-web/proxies): fix force ugprade

Introduced by a4d90e8aff0f816af48bc86093a3d629fd0f0aed

See xoa-support#3613

Forward options in `upgradeAppliance` effect.

julien-f committed:
fix(xo-web/proxies): fix force ugprade

Introduced by a4d90e8aff0f816af48bc86093a3d629fd0f0aed

See xoa-support#3613

Forward options in `upgradeAppliance` effect.

MathieuRA committed:
fixes

MathieuRA committed:
fixes

MathieuRA committed:
fixes

MathieuRA committed:
fixes

julien-f committed:
feat(xo-server/vm.set): dont switch to DMC when changing memory

Fixes #4983

julien-f committed:
feat(xo-server/vm.set): dont switch to DMC when changing memory

Fixes #4983

julien-f committed:
chore(CHANGELOG): update next

julien-f committed:
chore(CHANGELOG): update next

julien-f committed:
feat(@xen-orchestra/xapi): 0.6.1

julien-f committed:
feat(@xen-orchestra/xapi): 0.6.1

julien-f committed:
feat(@xen-orchestra/backups): 0.9.2

julien-f committed:
feat(@xen-orchestra/backups): 0.9.2

julien-f committed:
feat(xo-server-load-balancer): 0.5.0

julien-f committed:
feat(xo-server-load-balancer): 0.5.0

julien-f committed:
feat(xo-server): 5.78.3

julien-f committed:
feat(xo-server): 5.78.3

julien-f committed:
feat(xo-web): 5.80.1

julien-f committed:
feat(xo-web): 5.80.1

julien-f committed:
fix(backups/_copyDelta): dont pass extra params to watchStreamSize

Introduced by 9b1fbf0fb

julien-f committed:
fix(backups/_copyDelta): dont pass extra params to watchStreamSize

Introduced by 9b1fbf0fb

julien-f committed:
feat(@xen-orchestra/backups): 0.9.3

julien-f committed:
feat(@xen-orchestra/backups): 0.9.3

julien-f committed:
feat(xo-server): 5.78.4

julien-f committed:
feat(xo-server): 5.78.4

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,33 @@
+const RangeInput = withState<State, Props, Effects, Computed, ParentState, ParentEffects>(
+  {
+    initialState: () => ({
+      input: React.createRef(),
```
Why do you need this?

julien-f committed:
fix(CHANGELOG{,.unreleased}): move backup dev notes

Introduced by e7b846155

julien-f committed:
fix(CHANGELOG{,.unreleased}): move backup dev notes

Introduced by e7b846155

julien-f committed:
feat: release 5.57.1

julien-f committed:
feat: release 5.57.1

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,33 @@
+  input: React.RefObject<HTMLInputElement>
+}
+
+interface Props extends React.AllHTMLAttributes<HTMLInputElement> {
```
`Omit<React.ComponentPropsWithoutRef<'input'>, 'type'>`

https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase/#wrappingmirroring-a-html-element

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,33 @@
+  min: number
+  onChange: React.ChangeEventHandler<HTMLInputElement>
+  step: number
+  value: number
```
All these props are already inherited from `input`.

MathieuRA replied:
But by adding these props here, they become required.

julien-f committed:
feat(@xen-orchestra/proxy): 0.12.1

julien-f committed:
feat(@xen-orchestra/proxy): 0.12.1

julien-f committed:
chore(mixin): remove build step

julien-f committed:
chore(mixin): remove build step

julien-f committed:
feat(@xen-orchestra/mixin): 0.1.0

julien-f committed:
feat(@xen-orchestra/mixin): 0.1.0

julien-f committed:
feat(mixins): mixins shared between xo-proxy and xo-server

julien-f committed:
feat(mixins): mixins shared between xo-proxy and xo-server

julien-f committed:
chore(xo-server): use @xen-orchestra/mixins/hooks

julien-f committed:
chore(xo-server): use @xen-orchestra/mixins/hooks

julien-f committed:
chore(mixins): use PascalCase as they are classes

julien-f committed:
chore(mixins): use PascalCase as they are classes

julien-f committed:
feat(xo-server): use @xen-orchestra/mixins/Config

julien-f committed:
feat(xo-server): use @xen-orchestra/mixins/Config

MathieuRA committed:
remove reaclette

MathieuRA committed:
remove reaclette

MathieuRA committed:
remove reaclette

MathieuRA committed:
remove reaclette

julien-f committed:
chore(xo-server/xo-mixins): xo → app

- already used in some mixins
- used in xo-proxy

julien-f committed:
chore(xo-server/xo-mixins): xo → app

- already used in some mixins
- used in xo-proxy

badrAZ committed:
fix(@xen-orchestra/backups/_cleanVm): VHDs not correctly listed (#5720)

Introduced by 20f4c95

badrAZ committed:
fix(@xen-orchestra/backups/_cleanVm): VHDs not correctly listed (#5720)

Introduced by 20f4c95

julien-f committed:
fix(xo-server/math): fix ESLint directive

julien-f committed:
fix(xo-server/math): fix ESLint directive

julien-f committed:
fix(babel): dont ignore test files when linting

julien-f committed:
fix(babel): dont ignore test files when linting

julien-f committed:
chore(disposables): using → Disposable.use

julien-f committed:
chore(disposables): using → Disposable.use

julien-f committed:
chore: update promise-toolbox to 0.19.0

julien-f committed:
chore: update promise-toolbox to 0.19.0

julien-f committed:
fix(xapi/VDI_destroy): respect vdiDestroyRetryWhenInUse option

julien-f committed:
fix(xapi/VDI_destroy): respect vdiDestroyRetryWhenInUse option

julien-f committed:
chore(xapi/VDI_destroy): move retry condition in constructor

julien-f committed:
chore(xapi/VDI_destroy): move retry condition in constructor

julien-f committed:
feat(xapi): warn on retry

julien-f committed:
feat(xapi): warn on retry

julien-f committed:
fix(CHANGELOG.unreleased): @xen-orchestra/xapi

Introduced by 864946477

julien-f committed:
fix(CHANGELOG.unreleased): @xen-orchestra/xapi

Introduced by 864946477

badrAZ committed:
fix(xo-server/backup-ng): add slash between backup and remote ids (#5723)

This is symmetric to the parsing: https://github.com/vatesfr/xen-orchestra/blob/052aafd7cb48d20f427d1d3994c7b7bd388ce315/packages/xo-server/src/xo-mixins/backups-ng/index.js#L88-L94

badrAZ committed:
fix(xo-server/backup-ng): add slash between backup and remote ids (#5723)

This is symmetric to the parsing: https://github.com/vatesfr/xen-orchestra/blob/052aafd7cb48d20f427d1d3994c7b7bd388ce315/packages/xo-server/src/xo-mixins/backups-ng/index.js#L88-L94

badrAZ committed:
fix(fs/LocalHandler#_lock): correctly resolve path (#5726)

badrAZ committed:
fix(fs/LocalHandler#_lock): correctly resolve path (#5726)

julien-f committed:
feat(xo-server/getBackupNgLogs): expose proxyId

Follow up on b454b4dff

julien-f committed:
feat(xo-server/getBackupNgLogs): expose proxyId

Follow up on b454b4dff

julien-f committed:
fix(xapi/VM_create): default actions_after_{crash,reboot} is `restart`

See https://xapi-project.github.io/xen-api/classes/vm.html

`reboot` is not valid.

julien-f committed:
fix(xapi/VM_create): default actions_after_{crash,reboot} is `restart`

See https://xapi-project.github.io/xen-api/classes/vm.html

`reboot` is not valid.

julien-f committed:
fix(xapi/VM_destroy): correctly check *other* VM is not control domain

julien-f committed:
fix(xapi/VM_destroy): correctly check *other* VM is not control domain

julien-f committed:
fix(xo-server/api): fix config name entry verboseApiLogsOnErrors

julien-f committed:
fix(xo-server/api): fix config name entry verboseApiLogsOnErrors

julien-f committed:
fix(xo-server/recover-account): pass config as named param

Introduced by 7024c7d59

julien-f committed:
fix(xo-server/recover-account): pass config as named param

Introduced by 7024c7d59

MathieuRA committed:
remove required props

MathieuRA committed:
remove required props

MathieuRA committed:
remove required props

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,9 @@
+
+interface Props extends Omit<React.ComponentPropsWithoutRef<'input'>, 'type'> {}
+
+const RangeInput: React.FC<Props> = React.memo(props => <input {...props} type='range' />)
```
`React.FC` is discouraged, use the normal function typing instead.
https://react-typescript-cheatsheet.netlify.app/docs/basic/getting-started/function_components

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,9 @@
+import React from 'react'
+
+type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>
```
I don't get what you're doing here, it looks like you're redefining `Omit`.

pdonias commented on @xen-orchestra/lite/src/components/RangeInput.tsx:
```diff
@@ -0,0 +1,9 @@
+
+type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>
+
+interface Props extends Omit<React.ComponentPropsWithoutRef<'input'>, 'type'> {}
```
Why not simply:
```ts
type Props = Omit<React.ComponentPropsWithoutRef<'input'>, 'type'>
```

MathieuRA committed:
rebase

MathieuRA committed:
rebase

MathieuRA committed:
rebase

MathieuRA committed:
rebase xo-lite

MathieuRA committed:
remove unrelated change

MathieuRA committed:
remove unrelated change

MathieuRA committed:
return to previus working commit

MathieuRA committed:
Delete .npmignore

MathieuRA committed:
Delete .npmignore

MathieuRA committed:
Delete .npmignore

MathieuRA committed:
Delete .npmignore

MathieuRA committed:
remove unrelated change

MathieuRA committed:
unrelated change

MathieuRA committed:
unrelated change

MathieuRA committed:
Update manage_infrastructure.md

MathieuRA committed:
use normal function instead FC

MathieuRA committed:
Merge branch 'mathieu-xoLiteResizeConsole' of https://github.com/vatesfr/xen-orchestra into mathieu-xoLiteResizeConsole

pdonias commented on docs/manage_infrastructure.md:
```diff
@@ -625,4 +625,4 @@ xscontainer-prepare-vm -v <VM_UUID> -u <username>
 Because "prepare-vm" is not exposed outside of the Dom0 (yet?), we can't use Xen Orchestra to give you a one-click solution as of now.
-:::
\ No newline at end of file
+:::
```
Unrelated change.

MathieuRA replied:
I tried to remove this change but, he come back everytime. Dunno why

pdonias commented on @xen-orchestra/lite/src/components/TabConsole.tsx:
```diff
@@ -0,0 +1,53 @@
+        // With "scaleViewport", the canvas occupies all
+        // available space of its container.
+        // But when the size of the container is changed,
+        // the canvas size isn't updated
```
```suggestion
        // With "scaleViewport", the canvas occupies all available space of its
        // container. But when the size of the container is changed, the canvas
        // size isn't updated
```

MathieuRA committed:
Update @xen-orchestra/lite/src/components/TabConsole.tsx

Co-authored-by: Pierre Donias <pierre.donias@gmail.com>

MathieuRA committed:
unrelated change

MathieuRA committed:
unrelated change

MathieuRA committed:
Update manage_infrastructure.md

MathieuRA committed:
unrelated change

MathieuRA committed:
Merge branch 'mathieu-xoLiteResizeConsole' of https://github.com/vatesfr/xen-orchestra into mathieu-xoLiteResizeConsole

MathieuRA committed:
docs

MathieuRA committed:
Update manage_infrastructure.md

MathieuRA committed:
docs

MathieuRA committed:
Merge branch 'mathieu-xoLiteResizeConsole' of https://github.com/vatesfr/xen-orchestra into mathieu-xoLiteResizeConsole

MathieuRA committed:
docs

MathieuRA committed:
docs

MathieuRA committed:
move tabConsole into App/
